server/src/main/java/org/elasticsearch/index/mapper/DocumentMapper.java

```diff
@@ -93,13 +93,12 @@ public void validate(IndexSettings settings, boolean checkLimits) {
             }
         }
 
+        settings.getMode().validateMapping(mappingLookup);
```

Comment:
Validate index more before loading/validating source mode/loader.

The `nested fields` yaml tests validate the time series index mode doesn't support nested field type.
But without this the error returned is that synthetic source doesn't support nested field type.

The test can be adjusted, but logically it makes more sense to me that index mode reports the error than source mode, so that is why I moved `settings.getMode().validateMapping(mappingLookup);` before `sourceMapper().newSourceLoader(mapping());`.